File: Back/src/main/java/esprit/example/fegaac1/entities/Reservation.java
package esprit.example.fegaac1.entities;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_reservation;

    private LocalDate dateDebut;
    private LocalDate dateFin;
    private boolean approved ;

    @ManyToOne(cascade = CascadeType.ALL)
    Appartement  appartements;



}

File: Front/src/app/models/Reservation.ts
import { Appartement } from "./appartement";
import { User } from "./user";

export interface Reservation {
  id_reservation?: number;
  dateDebut: string;  
  dateFin: string;   
  approved: number;
  appartements?: Appartement;


    // Infos récupérées manuellement
  blocNom?: string;
  residenceNom?: string;
  paysNom?: string;

  user?: User;
} 
